orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/functions/ServiceKeyExpressionFunctionProvider.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 Pivotal, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.expressions.functions;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.TreeTraversingParser;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.jackson.OrcaObjectMapper;
+import com.netflix.spinnaker.orca.pipeline.expressions.ExpressionFunctionProvider;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import lombok.Data;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Predicate;
+
+@Component
+public class ServiceKeyExpressionFunctionProvider implements ExpressionFunctionProvider {
+  private static final String CREATE_SERVICE_KEY_STAGE_NAME = "createServiceKey";
+  private static final ObjectMapper objectMapper = OrcaObjectMapper.newInstance();
+
+  @Nullable
+  @Override
+  public String getNamespace() {
+    return null;
+  }
+
+  @NotNull
+  @Override
+  public Collection<FunctionDefinition> getFunctions() {
+    return Collections.singletonList(
+      new FunctionDefinition("serviceKey", Arrays.asList(
```

Comment:
my concern here is that this is the first function that is (sort of) provider specific. Perhaps worth prefixing it with `cf`, e.g. `cfServiceKey`? cc: @robzienert

Reply:
Also, don't forget to document it here: https://github.com/spinnaker/spinnaker.github.io/blob/master/reference/pipeline/expressions/index.md

Reply:
This was actually the point of to `getNamespace` method. So cloud provider spel helpers could be prefixed with the provider name. Then calling the method would look like so:

```
${#cloudfoundry.serviceKey(...)}
```

All this said, I don't like our provider specific logic in `orca-core` but that's currently the what we do for the other spel helpers today so I figure something out for think it's fine for now. This seems a perfect candidate later on for plugins?

Reply:
+1 on the namespacing, that was the intention.